lib/src/sdk/trace/exporters/collector_exporter.dart

```diff
@@ -13,14 +14,15 @@ import '../../proto/opentelemetry/proto/resource/v1/resource.pb.dart'
 import '../../proto/opentelemetry/proto/trace/v1/trace.pb.dart' as pb_trace;
 
 class CollectorExporter implements sdk.SpanExporter {
-  Uri uri;
-  http.Client client;
-  Map<String, String> headers;
+  final Uri uri;
+  final http.Client client;
+  final Map<String, String> headers;
   var _isShutdown = false;
 
-  CollectorExporter(this.uri, {http.Client httpClient, this.headers}) {
-    client = httpClient ?? http.Client();
-  }
+  CollectorExporter(this.uri,
+      {http.Client? httpClient, Map<String, String>? headers})
+      : client = httpClient ?? http.Client(),
+        headers = headers ?? {};
```

Comment:
I think you could shorten this up to:
```suggestion
  CollectorExporter(this.uri,
      {http.Client? httpClient, this.headers = const{}})
      : client = httpClient ?? http.Client();
```